spark-derived-feature-view/README.md

```diff
@@ -0,0 +1,85 @@
+# Overview
+
+This example shows how to use `DerivedFeatureView` to backfill the input dataset
+with extra features for offline training. It involves the following steps:
+
+1. Read a batch of historical purchase events from a file.
+
+   Each purchase event has the following fields:
+   - user_id, unique identifier of the user that made the purchase.
+   - item_id, unique identifier of the item that is purchased.
+   - item_count, number of items purcahsed.
+   - timestamp, time when this purchase is made.
+
+2. Read a batch of historical item price events from a file.
+
+   Each item price event has the following fields:
+   - item_id, unique identifier of the item.
+   - price, the new price of this item.
+   - timestamp, time when the new price is used for this item.
+
+3. For each purchase event, append the following two fields by joining with item
+   price events and performing over-window aggregation, with point-in-time
+   correctness in both operations.
+
+   - price, price of the item at the time this purchase is made.
+   - total_payment_last_two_minutes, total cost of purchases made by this
+     user in a 2-minute window that ends at the time this purchase is made.
+
+4. Output the batch of purchase events backfilled with the extra features to a
+   file.
+
+
+# Prerequisites
+
+Prerequisites for running this example:
+- Unix-like operating system (e.g. Linux, Mac OS X)
+
+# Step-By-Step Instructions
+
+Please execute the following commands under the `spark-derived-feature-view`
+folder to run this example.
+
+1. Build the Spark image with Feathub installed.
+
+   ```bash
+   $ docker build --rm -t feathub-spark -f ../docker/Dockerfile.spark ../docker
+   ```
+2. Start the Spark cluster.
+
+   ```bash
+   $ docker-compose up -d
+   ```
+
+   After the Spark cluster has started, you should be able to navigate to the
+   web UI at [localhost:8080](http://localhost:8080) to view the Spark dashboard.
+
+3. Run the FeatHub program to compute and output the extended purchase events to a file.
+   We submit Feathub program in a docker container instead of local workstation to avoid
+   the connection issue.
+
+   ```bash
+   $ docker exec spark bash -c "python3 /tmp/main.py"
```

Comment:
It seems inconsistent with the way we expect users to run FeatHub program in Spark client mode. Users should be able to start the Spark cluster and then let FeatHub driver program connect to it directly without having to log into a custom-built docker image, similar to how Spark's own quickstart works.